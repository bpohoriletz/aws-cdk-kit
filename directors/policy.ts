import { Construct } from 'constructs';
import * as iam from 'aws-cdk-lib/aws-iam';
import { IPolicyBuilder, IPolicyBuilderConstructor } from '../products/policy';

export default class PolicyDirector {
  private builder: IPolicyBuilder;

  constructor(builder: IPolicyBuilderConstructor) {
    this.builder = new builder();
  }

  constructPolicy(construct: Construct, id: string): iam.Policy {
    return new iam.Policy(construct, id, this.builder.getResult());
  }
}
