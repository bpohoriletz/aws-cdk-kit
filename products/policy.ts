import * as iam from 'aws-cdk-lib/aws-iam';

export class PolicyProduct implements iam.PolicyProps {}

export interface IPolicyBuilder {
  getResult(): PolicyProduct;
}

export interface IPolicyBuilderConstructor {
  new (): IPolicyBuilder;
}
