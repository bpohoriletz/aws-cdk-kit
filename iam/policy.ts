import * as cdk from 'aws-cdk-lib';
import * as iam from 'aws-cdk-lib/aws-iam';

import * as dev from './developer/policy';

export function createDeveloperPolicy(kind: string, tags: string[], stack: cdk.Stack): iam.ManagedPolicy[] | [] {
  switch (kind.toLowerCase()) {
    case 'ec2':
      return [dev.ec2Access(tags, stack)];
    default:
      return [];
  }
}
