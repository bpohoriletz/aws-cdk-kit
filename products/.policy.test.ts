import { PolicyProduct } from './policy';

describe('new PolicyProduct()', () => {
  let instance: PolicyProduct;

  beforeEach(() => {
    instance = new PolicyProduct();
  });

  test('has correct defaults', () => {
    expect(instance).toMatchSnapshot();
  });
});
