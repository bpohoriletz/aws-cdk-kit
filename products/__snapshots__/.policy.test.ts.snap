// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new PolicyProduct() has correct defaults 1`] = `PolicyProduct {}`;
